docs/manual/src/udl/ext_types_wrapped.md

```diff
@@ -57,13 +83,86 @@ Because `UniffiCustomTypeWrapper` is defined in each crate, this means you can w
 if they are not defined in your crate - see the 'wrapper_types' example which demonstrates
 how `serde_json::Value` can be wrapped.
 
-## Custom Foreign Wrappers
+## Bindings example
+
+Suppose you have Rust struct with recently renamed fields:
+
+```rust
+pub struct User {
+  full_name: String,
+  age: Int32,
+}
+```
+
+But your Kotlin code is still using the old field names:
+
+```kotlin
+data class LegacyUser(
+  var name: String,
+  var years_old: Int,
+}
+```
+
+We again start by declaring this type in our UDL via a `typedef` with a `Wrapped` attribute.
+
+```idl
+[Wrapped]
+typedef User LegacyUser;
+```
+
+This time however, we will update the Kotlin section of the `uniffi.toml` file:
+
+```toml
+[bindings.kotlin.wrappers.LegacyUser]
+# Name of the Kotlin class that will wrap `User`
+type_name = "LegacyUser"
+# Function that converts `User` to `LegacyUser`
+wrap = "wrapUser({})"
+# Function that converts `LegacyUser` to `User`
+unwrap = "unwrapUser({})"
+# Where we can import those functions from
+imports = [
+  "my.wrappers.package.wrapUser",
+  "my.wrappers.package.unwrapUser",
+]
+```
+
+`wrap` and `unwrap` specifies the code used to wrap/unwrap the type.  `{}`
+will be replaced with the value to be wrapped or unwrapped.
+
+`my.wrappers.package` would have code like this:
+
+```kotlin
+fun wrapUser(user: User) = LegacyUser(
```

Comment:
as Ben and I very briefly discussed, maybe leaning into "lift" and "lower" is actually what we should be doing here - those terms don't (afaik) come with existing baggage, and once explained are quite clear IMO. ie, it's a slightly steeper learning curve for new people, but there's less confusion once you've climbed it.

(OTOH, unless the semantics actually are identical to how uniffi already uses that term, the end result could be more confusion rather than less, which is clearly a terrible outcome - that's not clear to me yet)

Reply:
I have similar feels about lift and lower; as in: I like and prefer those terms, but worry about approachability.

> which is clearly a terrible outcome - that's not clear to me yet

😄